modules/core/src/main/scala/com.fortysevendeg.hood/github/GithubService.scala

```diff
@@ -95,6 +107,40 @@ object GithubService {
           .onError { case e => L.error(e)("Found error while accessing GitHub API.") }
       } yield result
 
+    def createStatus(
+        accessToken: String,
+        owner: String,
+        repository: String,
+        pullRequestNumber: Int,
+        state: GithubState,
+        targetUrl: Option[String],
+        description: String,
+        context: String
+    ): F[GHResponse[Status]] = {
+      val gh = Github(Some(accessToken))
+
+      (for {
+        pr <- EitherT(gh.pullRequests.get(owner, repository, pullRequestNumber).exec())
+        head <- EitherT.fromOption[F](
+          pr.result.head,
```

Comment:
This `head` looks very confusing, what is the type of `pr.result`?

Reply:
`head` refers to the head commit from the pullRequest (`result` is a GitHub4s `PullRequest` object). Statuses are associated to commits in a PR so I guessed the head commit would be the best option for this 😄